www/base/src/app/settings/service/settings.service.spec.coffee

```diff
@@ -0,0 +1,288 @@
+            ,
+                type:'bool'
+                name:'radio2'
+                default_value: false
+            ]
+        exceptionRun = ->
+            group_result = bbSettingsServiceProviderRef._addSettingsGroup(group)
+        expect(exceptionRun).toThrow()
+
+
+    it 'should merge groups when new group has item with no default value', inject (bbSettingsService) ->
+        localStorage.clear()
+        old_group =
+            name:'System'
+            caption: 'System related settings'
+            items:[]
+        new_group =
+            name:'System'
+            caption: 'System related settings'
+            items:[
+                type:'bool'
+                name:'checkbox_system'
+                default_value: false
+            ,
+                type:'bool'
+                name:'checkbox_system2'
+            ]
+        group_result = bbSettingsServiceProviderRef._mergeNewGroup(old_group, new_group)
+        expect(group_result).toEqual
+            name:'System'
+            caption: 'System related settings'
+            items:[
+                type:'bool'
+                name:'checkbox_system'
+                value: false
+                default_value: false
+            ,
+                type:'bool'
+                name:'checkbox_system2'
+                value: undefined
+            ]
+
+        
+    it 'should generate correct settings', inject (bbSettingsService) ->
+        groups = bbSettingsService.getSettingsGroups()
+        expect(groups['Release']).toEqual
+            name:'Release'
+            caption: 'Release related settings'
+            items:[
+                type:'bool'
+                name:'checkbox_release'
+                value:false
+                default_value: false
+            ,
+                type:'bool'
+                name:'checkbox_release2'
+                value: false
+                default_value: false
+            ,
+                type:'bool'
+                name:'checkbox_release3'
+                value:false
+                default_value: false
+            ,
+                type:'choices'
+                name:name
+                default_value: 'radio1'
+                value:'radio1'
+                answers: [
+                    { name: 'radio1' }
+                    { name: 'radio2' }
+                ]
+            ]
+
+    it 'should return correct setting', inject (bbSettingsService) ->
+        userSetting1 = bbSettingsService.getSetting('User.checkbox1')
+        userSetting2 = bbSettingsService.getSetting('User.whatever')
+        userSetting3 = bbSettingsService.getSetting('UserAA.User_checkbox1')
+        expect(userSetting1).toBeDefined()
+        expect(userSetting2).toBeUndefined()
+        expect(userSetting3).toBeUndefined()
+
+    it 'should save correct settings', inject (bbSettingsService) ->
+        checkbox = bbSettingsService.getSetting('User.checkbox1')
+        expect(checkbox.value).toBeFalsy()
```

Comment:
expect(checkbox.value).toBe(false) is more strict.
if checkbox.value is undefined, expect(checkbox.value).toBeFalsy() is true